src/v/cluster/rm_stm.cc

```diff
@@ -1182,6 +1181,7 @@ ss::future<result<kafka_result>> rm_stm::do_idempotent_replicate(
         req_ptr->set_value<ret_t>(result.error());
         co_return result.error();
     }
+    units.return_all();
```

Comment:
it looks like this release is still to early as the step down happens on line 1129 outside of `do_idempotent_replicate` should we keep the units alive there ?

Reply:
I think it does? do_idempotent_replicate() takes a reference to the units as the input parameter.

Reply:
now i see it, it is a bit confusing